### PR TITLE
Feat/m2 506 default screen

### PR DESCRIPTION
**Pull request by BamMironov**
- Added logic responsible for rendering default route.
- Added logic responsible for persisting particular routes.
- Split the `@screens` layer into segments for better maintenance.

### CODE REVIEW

File: src/screens/lib/navigationService.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { createSecureStorage } from '@shared/lib';
+
+import { InitialRoute } from './types';
+
+const storage = createSecureStorage('navigation-storage');
+
+function SessionService() {
```
Comment (anq83) [inline src/screens/lib/navigationService.ts:7]: Shouldn't it be named the same as file name?

File: src/screens/lib/navigationService.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  return {
+    setInitialRoute(route: InitialRoute) {
```
Comment (anq83) [inline src/screens/lib/navigationService.ts:9]: Can we rename it to setStartupRoute?

File: src/screens/lib/types.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { ScreenParams, ScreenRoute } from '../config';
+
+export type InitialRoute = {
```
Comment (anq83) [inline src/screens/lib/types.ts:3]: Maybe StartupRoute?

File: src/screens/lib/navigationService.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      storage.set('initialRoute', JSON.stringify(route));
+    },
+
+    clearInitialRoute() {
+      storage.delete('initialRoute');
+    },
+
+    getInitialRoute() {
```
Comment (anq83) [inline src/screens/lib/navigationService.ts:17]: the same

File: src/screens/ui/RootNavigator.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            color: colors.white,
+          },
+        }}
+        component={AboutScreen}
+      />
+
+      <Stack.Screen
+        name="Applets"
+        options={{
+          title: 'Applets',
```
Comment (anq83) [inline src/screens/ui/RootNavigator.tsx:86]: The Title should be "Hi [User Name]"

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/screens/lib/navigationService.ts (hunk 1 above)
Comment (BamMironov) [reply src/screens/lib/navigationService.ts:7]: Correct

File: src/screens/lib/navigationService.ts (hunk 2 above)
Comment (BamMironov) [reply src/screens/lib/navigationService.ts:9]: Not exactly. "Initial route" has been coined by react-navigation so I just follow their API naming

File: src/screens/lib/types.ts (hunk 3 above)
Comment (BamMironov) [reply src/screens/lib/types.ts:3]: Please see above

File: src/screens/lib/navigationService.ts (hunk 4 above)
Comment (BamMironov) [reply src/screens/lib/navigationService.ts:17]: Please see above

File: src/screens/ui/RootNavigator.tsx (hunk 5 above)
Comment (BamMironov) [reply src/screens/ui/RootNavigator.tsx:86]: I can do but I don't think it is related to my PR because it was not me who worked on this screen

**Review by anq83 — APPROVED** [review]

**Pull request merged**